extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -1646,3 +1656,49 @@ export class Bind {
     }
   }
 }
+
+/** @private @const {!Object<string, boolean>} */
+const CAN_BE_DISABLED = {
+  'FIELDSET': true,
+  'INPUT': true,
+  'SELECT': true,
+  'TEXTAREA': true,
+};
+
+/**
+ * Returns whether the element's property affects the value of a form field.
+ * @param {string} tagName
+ * @param {string} property
+ * @return {boolean}
+ */
+function doesPropertyAffectFormValue(tagName, property) {
+  switch (property) {
+    case 'disabled':
```

Comment:
Should we consider [hidden] attribute as well in this case then? Feel free to left a TODO and resolve in another PR.

Note that `hidden` attribute can be mutated on the ancestry of a form field.

Reply:
Ugh right. So there are:

- `disabled` for form fields and fieldsets
- `hidden` for all essentially all elements
- `type` for `<input>`

I am still leaning towards using a mutation observer on the form for these, and leave `AmpEvents.FORM_VALUE_CHANGE` as close to the native `InputEvent` as possible. (It's also semantically cleaner as Will mentioned)

Thoughts? @choumx @GoTcWang

Reply:
Yup that also works.
I'll left it to Will to make the call.

Reply:
SGTM